### PR TITLE
 added tencent bugly and also updated old DexProtector

### PR DESCRIPTION
**Pull request by Jasi2169**

### CODE REVIEW

File: apkid/rules/apk/packers.yara [hunk 1]
```diff
@@ -114,6 +115,23 @@ rule dexprotector : packer
...
 
+rule tencentbugly : packer
+{
+ /**
+ * Tencent Bugly v1.x/v2.x (MultiDex Packer)
+ **/
+ 
+  meta:
+    author = "Jasi2169"
+    description = "Tencent Bugly"
```
Comment (CalebFenton) [inline apkid/rules/apk/packers.yara]: Could you add a sample hash here, please. Also, is there a website for this? How is this different from the normal tencent packers?

File: apkid/rules/apk/packers.yara [hunk 2]
```diff
@@ -114,6 +115,23 @@ rule dexprotector : packer
...
+    
+  strings:
+    $main_lib = "libBugly.so"
+    
+  condition:
+    apk and $main_lib
```
Comment (CalebFenton) [inline apkid/rules/apk/packers.yara]: This seems a little week. Any way to fingerprint the actual unpacking code? Also, I think `apk` is causing the rules to fail to compile, hence the failing build. I think it's supposed to be `is_apk`. You can test the rules yourself by running `./prep_release.py`.

**Comment by CalebFenton** [comment]
Thanks for stepping up and trying to add some new rules.

File: apkid/rules/apk/packers.yara (hunk 1 above)
Comment (enovella) [reply apkid/rules/apk/packers.yara]: Isn't Bugly a crash report lib? https://mvnrepository.com/artifact/com.tencent.bugly/crashreport

Comment (Jasi2169) [reply apkid/rules/apk/packers.yara]: i am at wedding i will add extra stuff,i will send you sample caleb plz email me at jaspreetsinghplayer[at]gmail[dot]com i send you sample it is multidex packer includes tinker maybe that adds overlay,the developer of app will not talk about it but there is chinese forums which does shows i need to lock after wedding,see yeah

**Comment by CalebFenton** [comment]
Tencent Bugly is actually just an error reporting library often included with tencent mobile protect. I found some samples and added rules for it here https://github.com/rednaga/APKiD/commit/b9a1073f1ebf9f1edb98d4bc16fe653778a1a3ac

**Pull request closed without merging**